owncloudApp/src/androidTest/java/com/owncloud/android/shares/domain/privateShares/DeletePrivateShareTest.kt

```diff
@@ -0,0 +1,292 @@
+            remoteId = 20,
+            shareType = ShareType.GROUP.value,
+            shareWith = "family",
+            path = "/Documents",
+            permissions = Permissions.READ_PERMISSIONS.value,
+            isFolder = true,
+            sharedWithDisplayName = "Family"
+        )
+    )
+
+    private val capabilitiesLiveData = MutableLiveData<Resource<OCCapability>>()
+    private val privateSharesLiveData = MutableLiveData<Resource<List<OCShare>>>()
+
+    private val ocCapabilityViewModel = mock(OCCapabilityViewModel::class.java)
+    private val ocShareViewModel = mock(OCShareViewModel::class.java)
+
+    companion object {
+        private val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
+        private val account = Account("admin", "owncloud")
+
+        @BeforeClass
+        @JvmStatic
+        fun init() {
+            addAccount()
+        }
+
+        @AfterClass
+        @JvmStatic
+        fun cleanUp() {
+            AccountsManager.deleteAllAccounts(targetContext)
+        }
+
+        private fun addAccount() {
+            // obtaining an AccountManager instance
+            val accountManager = AccountManager.get(targetContext)
+
+            accountManager.addAccountExplicitly(account, "a", null)
+
+            // include account version, user, server version and token with the new account
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_OC_VERSION,
+                OwnCloudVersion("10.2").toString()
+            )
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_OC_BASE_URL,
+                "serverUrl:port"
+            )
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_DISPLAY_NAME,
+                "admin"
+            )
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_OC_ACCOUNT_VERSION,
+                "1"
+            )
+
+            accountManager.setAuthToken(
+                account,
+                AccountAuthenticator.KEY_AUTH_TOKEN_TYPE,
+                "AUTH_TOKEN"
+            )
+        }
+    }
+
+    @Before
+    fun setUp() {
+        val intent = Mockito.spy(Intent::class.java)
+
+        file = TestUtil.createFile("image.jpg")
+
+        `when`(intent.getParcelableExtra(FileActivity.EXTRA_FILE) as? Parcelable).thenReturn(file)
+        intent.putExtra(FileActivity.EXTRA_FILE, file)
+
+        `when`(ocCapabilityViewModel.getCapabilityForAccount(false)).thenReturn(capabilitiesLiveData)
+        `when`(ocCapabilityViewModel.getCapabilityForAccount(true)).thenReturn(capabilitiesLiveData)
+        `when`(ocShareViewModel.getPrivateShares(file.remotePath)).thenReturn(privateSharesLiveData)
+        `when`(ocShareViewModel.getPublicShares(file.remotePath)).thenReturn(MutableLiveData())
+
+        stopKoin()
```

Comment:
I would create a method with annotation @After to put 'stopKoin()' there instead.

Reply:
I can create it to stop Koin for the last executed test but this line would be still needed because since instrumented tests are launched on a device, the android application class is instantiated and it causes Koin initialization as a side effect Therefore, we need to stop it here before starting koin again, otherwise I get the exception below:


`org.koin.core.error.KoinAppAlreadyStartedException: A Koin Application has already been started`

Reply:
You can create a full Koin tree to be used only in tests and load it in the @Begin methods with ` StandAloneContext.loadKoinModules` so that if overrides the full app module(s), then keep the @Begin/@After symmetry.